plugins/default/lib/openshift.py

```diff
@@ -44,7 +44,51 @@ def exec_in_pods(project, pods, cmd):
     return _exec_in_pods(oc, project, pods, cmd)
 
 
+def _exec_in_pod(oc, project, pod, cmd):
+    return oc("-n", project, "exec", pod, "--", *cmd)
+
+
+def exec_in_pod(project, pod, cmd):
+    return _exec_in_pod(oc, project, pod, cmd)
+
+
+def _exec_in_pod_container(oc, project, pod, container, cmd):
+    return oc("-n", project, "exec", pod, "-c", container, "--", *cmd)
+
+
+def exec_in_pod_container(project, pod, container, cmd):
+    return _exec_in_pod_container(oc, project, pod, container, cmd)
+
+
+def _get_running_pod_containers(oc, project, selector=None):
+    args = ("-n", project, "get", "pods", "--show-all=false", "-o", "json")
+
+    if selector:
+        args += ("--selector=" + ",".join(selector),)
+
+    pods = json.loads(oc(*args))["items"]
+
+    result = []
+    for p in pods:
+        if p["status"]["phase"] == "Running":
+            for c in p["spec"]["containers"]:
+                result.append((p["metadata"]["name"], c["name"]))
+
+    return result
+
+    # Technically, all of the above could be done in one shot, but its too hard to read
+    # return list(itertools.chain.from_iterable(
+    #     [[(p["metadata"]["name"], c["name"]) for c in p["spec"]["containers"]]
+    #         for p in pods if p["status"]["phase"] == "Running"]
+    # ))
+
+
+def get_running_pod_containers(project, selector=None):
+    return _get_running_pod_containers(oc, project, selector)
+
+
 def get_project():
+    # return oc("project", "-q").rstrip("\n")
```

Comment:
Yeah, I keep putting that back in so I can easily test the python from my mac without having to put code back into the pod.  I'll remove it before merging.